nncf/quantization/passes.py

```diff
@@ -0,0 +1,69 @@
+"""
+ Copyright (c) 2023 Intel Corporation
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+      http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+
+import collections
+
+from nncf.common.graph.graph import NNCFGraph
+from nncf.common.graph.operator_metatypes import ShapeOfMetatype
+
+
+def transform_to_inference_graph(nncf_graph: NNCFGraph) -> NNCFGraph:
+    """
+    This method contains pipeline of the passes that uses to provide inference graph without constant flows.
+
+    :param nncf_graph: NNCFGraph instance for the transformation.
+    :return: NNCFGraph in the inference style.
+    """
+    inference_nncf_graph = remove_shape_of_subgraphs(nncf_graph)
+    return inference_nncf_graph
+
+
+def remove_shape_of_subgraphs(nncf_graph: NNCFGraph) -> NNCFGraph:
+    """
+    Removes the ShapeOf subgraphs from the provided NNCFGraph instance.
+
+    :param nncf_graph: NNCFGraph instance for the transformation.
+    :return: NNCFGraph without ShapeOf subgraphs.
+    """
+    nodes_to_drop = set()
+    shape_of_nodes = []
+    infer_nodes = []
+
+    nodes_queue = collections.deque()
+    nodes_queue.extend(nncf_graph.get_input_nodes())
```

Comment:
Minor improvements. Up to you
```suggestion
    nodes_queue = collections.deque(nncf_graph.get_input_nodes())
```